CodePushPackage.m

```diff
@@ -149,50 +149,51 @@ + (NSString *)getPackageFolderPath:(NSString *)packageHash
 }
 
 + (void)downloadPackage:(NSDictionary *)updatePackage
-                            error:(NSError **)error
+       progressCallback:(void (^)(long, long))progressCallback
+           doneCallback:(void (^)())doneCallback
+           failCallback:(void (^)(NSError *err))failCallback
 {
     NSString *packageFolderPath = [self getPackageFolderPath:updatePackage[@"packageHash"]];
+    NSError *error = nil;
     
     if (![[NSFileManager defaultManager] fileExistsAtPath:packageFolderPath]) {
         [[NSFileManager defaultManager] createDirectoryAtPath:packageFolderPath
                                   withIntermediateDirectories:YES
                                                    attributes:nil
-                                                        error:error];
+                                                        error:&error];
     }
     
-    if (*error) {
-        return;
-    }
-    
-    NSURL *url = [[NSURL alloc] initWithString:updatePackage[@"downloadUrl"]];
-    NSString *updateContents = [[NSString alloc] initWithContentsOfURL:url
-                                                              encoding:NSUTF8StringEncoding
-                                                                 error:error];
-    if (*error) {
-        return;
-    }
-    
-    [updateContents writeToFile:[packageFolderPath stringByAppendingPathComponent:@"app.jsbundle"]
-                     atomically:YES
-                       encoding:NSUTF8StringEncoding
-                          error:error];
-    if (*error) {
-        return;
-    }
-    
-    NSData *updateSerializedData = [NSJSONSerialization dataWithJSONObject:updatePackage
-                                                                   options:0
-                                                                     error:error];
-    
-    if (*error) {
-        return;
-    }
+    if (error) {
+        return failCallback(error);
+    }
+    
+    NSString *updateBundleFileName = [packageFolderPath stringByAppendingPathComponent:@"app.jsbundle"];
```

Comment:
This constant (@"app.jsbundle") is duplicated in CodePush.m.  Can you deduplicate?